core/domain/state_domain.py

```diff
@@ -540,6 +579,85 @@ def from_dict(cls, interaction_dict):
             [Hint.from_dict(h) for h in interaction_dict['hints']],
             solution_dict)
 
+    def to_proto(self):
```

Comment:
Suggest to_android_proto & ditto elsewhere (e.g. to_android_voiceover_proto) (i.e. 'android' is always after 'to_') to contextualize that these are Android-specific conversion methods. That should help minimize people inadvertently using this for non-Android cases.

Reply:
Yup, let's do what @BenHenning  says. (I'm discussing with him atm.)

Reply:
Done